snapfaas/src/fs/mod.rs

```diff
@@ -300,12 +300,49 @@ pub struct Gate {
     object_id: UID,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum HttpVerb { GET, POST, PUT, DELETE }
+
+impl From<HttpVerb> for reqwest::Method {
+    fn from(verb: HttpVerb) -> Self {
+        match verb {
+            HttpVerb::GET    => reqwest::Method::GET,
+            HttpVerb::POST   => reqwest::Method::POST,
+            HttpVerb::PUT    => reqwest::Method::PUT,
+            HttpVerb::DELETE => reqwest::Method::DELETE,
+        }
+    }
+}
+
+impl From<reqwest::Method> for HttpVerb {
+    fn from(method: reqwest::Method) -> Self {
+        match method {
+            reqwest::Method::GET    => HttpVerb::GET,
+            reqwest::Method::POST   => HttpVerb::POST,
+            reqwest::Method::PUT    => HttpVerb::PUT,
+            reqwest::Method::DELETE => HttpVerb::DELETE,
+            _ => panic!("Request method {} not supported", method)
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
```

Comment:
We can store headers, token, url, verb in another struct and store it at the location keyed by `object_id`. We define a default set of headers and we the default set if no header is provided at creation.

Reply:
Fixed. Added another data structure `ServiceInfo`